packages/interactivity/src/directives.js

```diff
@@ -226,7 +228,12 @@ export default () => {
 						// A `false` value is different from the attribute not being
 						// present, so we can't remove it.
 						// We follow Preact's logic: https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L131C24-L136
-						if ( result === false && attribute[ 4 ] !== '-' ) {
+						if (
+							( result === false ||
+								result === undefined ||
+								result === null ) &&
+							attribute[ 4 ] !== '-'
+						) {
```

Comment:
This needs to be moved to its own PR and reviewed to match [Preact's logic](https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L131C24-L136)